dist/js/ui-components.js

```diff
@@ -623,6 +657,78 @@
 
 	module.exports = "<div class=\"toolbar-pf-view-selector form-group\">\n  <ul class=\"list-inline\">\n    <li ng-repeat=\"item in vm.toolbarViews\" ng-class=\"{active: item.selected}\">\n      <a href=\"javascript:void(0)\"\n         title=\"{{item.title}}\"\n         id=\"{{item.id}}\"\n         data-url=\"{{item.url}}\"\n         data-url_parms=\"{{item.url_parms}}\"\n         ng-click=\"vm.onItemClick({item: item})\"\n         name=\"{{item.name}}\">\n        <i class=\"{{item.icon}}\" style=\"\"></i>\n      </a>\n    </li>\n  </ul>\n</div>\n"
 
+/***/ },
+/* 30 */
+/***/ function(module, exports, __webpack_require__) {
+
+	"use strict";
+	var stateAndView_1 = __webpack_require__(31);
+	Object.defineProperty(exports, "__esModule", { value: true });
+	exports.default = function (module) {
+	    module.filter('filterByStateAndView', stateAndView_1.default.filter);
+	};
+
+
+/***/ },
+/* 31 */
+/***/ function(module, exports, __webpack_require__) {
+
+	"use strict";
+	var toolbarType_1 = __webpack_require__(32);
+	/**
+	 * @memberof miqStaticAssets
+	 * @ngdoc filter
+	 * @name StateAndViewFilter
+	 */
+	var StateAndViewFilter = (function () {
+	    function StateAndViewFilter() {
+	    }
+	    /**
+	     * Filter items based on type and id. Type has to be {@link miqStaticAssets.ToolbarType.BUTTON_TWO_STATE} and id
+	     * can't start with `view_`.
+	     * @memberof StateAndView
+	     * @function filter
+	     * @returns {function(any): any}
+	     */
+	    StateAndViewFilter.filter = function () {
+	        return function (toolbarItems) {
+	            return toolbarItems.filter(function (toolbarItem) {
+	                return toolbarItem.type === toolbarType_1.ToolbarType.BUTTON_TWO_STATE && toolbarItem.id.indexOf('view_') === -1;
+	            });
+	        };
+	    };
+	    return StateAndViewFilter;
+	}());
+	Object.defineProperty(exports, "__esModule", { value: true });
+	exports.default = StateAndViewFilter;
+
+
+/***/ },
+/* 32 */
+/***/ function(module, exports) {
+
+	"use strict";
+	/**
+	 * Enum for toolbar types. It holds string value of item's type, so accessing these values can be called as:
+	 * ```Javascript
+	 * ToolbarType.BUTTON;
+	 * ToolbarType.BUTTON_TWO_STATE;
+	 * ToolbarType.BUTTON_SELECT;
+	 * ToolbarType.CUSTOM;
+	 * ```
+	 * It will return string interpretation of toolbarType.
+	 * @memberof miqStaticAssets
+	 * @ngdoc enum
+	 * @name ToolbarType
+	 */
+	var ToolbarType = (function () {
```

Comment:
@karelhala We may have a problem here, have you tested this please?

The ts code looks good and I'd merge .. but seems like in the compiled JS, `ToolbarType.BUTTON_TWO_STATE` will always be `undefined` (and the others..)